hal/source/arg_pack.cpp

```diff
@@ -135,6 +166,42 @@ bool hal_argpack_t::build(const hal_arg_t *args, uint32_t num_args) {
   return true;
 }
 
+hal_argpack_t::~hal_argpack_t() {
+  if (pack) {
+    std::free(pack);
+  }
+}
+
+void hal_argpack_t::clear() {
+  write_point = 0;
+  local_current_ptr = local_start;
+}
+
+bool hal_argpack_t::expand(size_t num_bytes) {
+  constexpr uint32_t alignment = 128;
+  constexpr uint32_t chunk_size = 128;
+
+  if (!pack || (write_point + num_bytes > pack_alloc_size)) {
+    // Allocate in chunks so we don't continually reallocate
+    pack_alloc_size =
+        ((write_point + num_bytes) + chunk_size - 1) / chunk_size * chunk_size;
+    uint8_t *replacement_pack = reinterpret_cast<uint8_t *>(
```

Comment:
Here, `static_cast` and `reinterpret_cast` are equivalent, in which case `static_cast` is probably better as it provides more checks that the cast makes sense.